hnn_core/network.py

```diff
@@ -400,6 +420,57 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
+    def _reset_cell_positions(self):
+        pos = _create_cell_coords(n_pyr_x=self._N_pyr_x, n_pyr_y=self._N_pyr_y,
+                                  zdiff=self._layer_separation,
+                                  inplane_distance=self._inplane_distance)
+        for key in pos.keys():
+            self.pos_dict[key] = pos[key]
+
+    @property
+    def inplane_distance(self):
+        return self._inplane_distance
+
+    @inplane_distance.setter
+    def inplane_distance(self, new_dist):
+        if len(self.external_drives) > 0:
+            raise RuntimeError('In-plane distance between network cells can '
+                               'only be adjusted before external drives are '
+                               f'attached; have {len(self.external_drives)} '
+                               'drives.')
+        _validate_type(new_dist, (float, int), 'new distance')
+        if not new_dist > 0.:
+            raise ValueError('In-plane distance must be positive definite, '
+                             f'got: {new_dist}')
+
+        # Adjust the lamtha of all cells; note that this would have no effect
+        # on external drives, since the event times there are calculated when
+        # the drives are attached.
+        for conn in self.connectivity:
+            conn['nc_dict']['lamtha'] *= new_dist / self._inplane_distance
```

Comment:
feels a bit like a hack. If a user inspected the `net.connectivity` they would be surprised that `lamtha` changed. And also it doesn't work for drives. I think we should pass `inplane_distance` here so it's applied uniformly and doesn't change `lamtha`:

https://github.com/jonescompneurolab/hnn-core/blob/0d40aae54abc313277089b750654637a4c5826d9/hnn_core/network_builder.py#L501